bin-src/lib/getOptions.ts

```diff
@@ -181,70 +150,28 @@ export default function getOptions({ argv, env, flags, log, packageJson }: Conte
-  if (scriptName === undefined && !exec && !noStart && !storybookUrl && !port) {
-    if (storybookBuildDir) {
-      return { ...options, noStart: true, useTunnel: false };
-    }
-    const { scripts } = packageJson;
-    if (typeof buildScriptName !== 'string') {
-      buildScriptName = 'build-storybook';
-      if (!scripts[buildScriptName]) {
-        const [key] =
-          Object.entries(scripts as Record<string, string>).find(([, script]) =>
-            script.startsWith('build-storybook')
-          ) || [];
-        if (key) buildScriptName = key;
-      }
-    }
-    if (scripts && buildScriptName && scripts[buildScriptName]) {
-      return { ...options, noStart: true, useTunnel: false, buildScriptName };
+  // Build Storybook
+  if (storybookBuildDir) {
+    return { ...options };
+  }
+  const { scripts } = packageJson;
+  if (typeof buildScriptName !== 'string') {
+    buildScriptName = 'build-storybook';
+    if (!scripts[buildScriptName]) {
+      const [key] =
+        Object.entries(scripts as Record<string, string>).find(([, script]) =>
+          script.startsWith('build-storybook')
+        ) || [];
+      if (key) buildScriptName = key;
     }
-    throw new Error(missingBuildScriptName(buildScriptName));
   }
+  if (scripts && buildScriptName && scripts[buildScriptName]) {
+    return { ...options, buildScriptName };
+  }
+  throw new Error(missingBuildScriptName(buildScriptName));
 
   // TurboSnap requires a static build with a webpack stats file.
   if (options.onlyChanged) throw new Error(invalidOnlyChanged());
 
-  // Start Storybook on localhost and generate the URL to it
-  if (!storybookUrl) {
-    if (exec && !port) {
-      throw new Error(missingStorybookPort());
-    }
-
-    if (!exec && (!port || !noStart)) {
-      // If you don't provide a port or we need to start the command, let's look up the script for it
-      scriptName = scriptName || 'storybook';
-      const storybookScript = packageJson.scripts && packageJson.scripts[scriptName];
-
-      if (!storybookScript) {
-        throw new Error(missingScriptName(scriptName));
-      }
-
-      options.https =
-        options.https ||
-        (getStorybookConfiguration(storybookScript, '--https') && {
-          cert: resolveHomeDir(getStorybookConfiguration(storybookScript, '--ssl-cert')),
-          key: resolveHomeDir(getStorybookConfiguration(storybookScript, '--ssl-key')),
-          ca: resolveHomeDir(getStorybookConfiguration(storybookScript, '--ssl-ca')),
-        });
-
-      port = port || getStorybookConfiguration(storybookScript, '-p', '--port');
-      if (!port) {
-        throw new Error(unknownStorybookPort(scriptName));
-      }
-
-      if (log) log.info('', inferredOptions({ scriptName, port }));
-    }
-
-    storybookUrl = `${options.https ? 'https' : 'http'}://localhost:${port}`;
-  }
-
-  return {
-    ...options,
-    noStart,
-    useTunnel: true,
-    url: storybookUrl,
-    scriptName,
-  };
+  return options;
```

Comment:
These lines are unreachable after the `throw` above.

The check for `onlyChanged` was there because it's incompatible with starting Storybook rather than using a built Storybook. We can remove this check because we already check for the existence of the stats file and will just bail with `missingStatsFile` if it doesn't exist, and continue running the build regardless.